components/product_notices_modal/product_notices_modal.tsx

```diff
@@ -0,0 +1,284 @@
+
+import Markdown from 'components/markdown';
+import GenericModal from 'components/generic_modal';
+import NextIcon from 'components/widgets/icons/fa_next_icon';
+import PreviousIcon from 'components/widgets/icons/fa_previous_icon';
+import AdminEyeIcon from 'components/widgets/icons/admin_eye_icon';
+
+import {isDesktopApp, getDesktopVersion} from 'utils/user_agent';
+
+import './product_notices_modal.scss';
+
+type Props = {
+    version: string;
+    currentTeamId: string;
+    socketStatus: WebsocketStatus;
+    actions: {
+        getInProductNotices: (teamId: string, client: string, clientVersion: string) => Promise<{
+            data: ProductNotices;
+        }>;
+        updateNoticeAsViewed: (noticeIds: string[]) => Promise<Record<string, unknown>>;
+    };
+}
+
+type State = {
+    presentNoticeIndex: number;
+    noticesData: ProductNotices;
+}
+
+export default class ProductNoticesModal extends React.PureComponent<Props, State> {
+    clearDataTimer?: number;
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            presentNoticeIndex: 0,
+            noticesData: [],
+        };
+        this.clearDataTimer = undefined;
+    }
+
+    public componentDidMount() {
+        this.fetchNoticesData();
+    }
+
+    public componentDidUpdate(prevProps: Props) {
+        const presentSocketState = this.props.socketStatus;
+        const prevSocketState = prevProps.socketStatus;
+        if (presentSocketState.connected && !prevSocketState.connected && prevSocketState.lastConnectAt) {
+            const presentTime = Date.now();
+            const previousSocketConnectDate = new Date(prevSocketState.lastConnectAt).getDate();
+            const presentDate = new Date(presentTime).getDate();
+            if (presentDate !== previousSocketConnectDate && presentTime > prevSocketState.lastConnectAt) {
+                this.fetchNoticesData();
+            }
+        }
+    }
+
+    public componentWillUnmount() {
+        clearTimeout(this.clearDataTimer);
+    }
+
+    private async fetchNoticesData() {
+        const {version, currentTeamId} = this.props;
+        let client = 'web';
+        let clientVersion = version;
+        if (isDesktopApp()) {
+            client = 'desktop';
+            clientVersion = getDesktopVersion();
+        }
+
+        const {data} = await this.props.actions.getInProductNotices(currentTeamId, client, clientVersion);
+        if (data) {
+            this.setState({
+                noticesData: data,
+            });
+            if (data.length) {
+                const presentNoticeInfo = this.state.noticesData[this.state.presentNoticeIndex];
+                this.props.actions.updateNoticeAsViewed([presentNoticeInfo.id]);
+            }
+        }
+    }
+
+    private confirmButtonText(presentNoticeInfo: ProductNotice) {
+        const noOfNotices = this.state.noticesData.length;
+
+        if (noOfNotices === 1 && presentNoticeInfo.actionText) {
```

Comment:
Does `actiontext` only exist when there is just one notice?

Reply:
No, But if it is just one notice then instead of showing `next` or `done` we show whatever the text is i.e like Update